### DISCUSSION TITLE
In Visual Studio Solution, how not to copy any dll's in vcpkg into binary's folder?

### DISCUSSION THREAD

**Discussion by playgithub** [microsoft/vcpkg#19128] [Ideas]
By default, e.g. it'll copy all depending dll's into `path\to\vs_solution\x64\Debug`.
Any option to not copy the dll's?

**Comment by JackBoosY** [comment]
Without copy the dependencies, your program will not run or be debug successfully.

**Comment by playgithub** [comment]
@JackBoosY
The system environemnt variable `PATH` can be used to find the dll's.

**Comment by JackBoosY** [comment]
Usually, the program will find the folder where the program is located, the folder in the `PATH` and whether there are dependent libraries in the `system32` folder . Users use these dependent libraries and need to **publish** them at the same time, so I think it is doing well now.

**Comment by playgithub** [comment]
It's ok, just to see if it can be better.
It's useful to know the depent dll's when publishing. But it's not necessary when developing. After `vcpkg integrate install`,
`.lib` can be linked automatically, so why not `.dll` be found automatically. But in this way depent dll's become implicit.

Is it better to have an option of vcpkg in VS project  setting that contorls wheather to copy depent dll's?
Or is it better to keep simple as it is now?

**Comment by JackBoosY** [comment]
Maybe we can add a option to control this. @strega-nil Do you agree with it?

**Comment by strega-nil** [comment]
Yeah, sounds reasonable. I'll talk about it at next weeks meeting.

**Comment by strega-nil** [comment]
Assigning @ras0219 since this is apparently a thing already.

**Comment by Pietrijew** [comment]
Alternatively you can just set some additional paths to DLL folders for particular project within the solution. To do so, just go to "Project Properties" -> "Debugging" -> then set the "Environment" value to something like "PATH=%PATH%;C:\DEV\vcpkg\installed\x64-windows\bin;"
You can also set a system variable called for example "VCPKG_X64_INSTALLED" with value "C:\DEV\vcpkg\installed\x64-windows" - then the "Environment" setting in VS would be like "PATH=%PATH%;$(VCPKG_X64_INSTALLED)\bin;"

**Reply by playgithub** [reply]
In favor of a automatic way when possible
